android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java

```diff
@@ -3,44 +3,192 @@
 import android.app.Activity;
 import android.content.Context;
 import android.content.Intent;
-import android.content.IntentFilter;
-import android.content.BroadcastReceiver;
-import android.os.BatteryManager;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.Settings;
 import android.view.WindowManager;
 
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
-import com.facebook.react.bridge.WritableMap;
-import com.facebook.react.bridge.WritableNativeMap;
 import com.facebook.react.bridge.ReactMethod;
-import com.facebook.react.bridge.ReadableArray;
 import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.WritableNativeMap;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
-import com.facebook.react.bridge.LifecycleEventListener;
 
-import javax.annotation.Nullable;
 
 public class ScreenBrightnessModule extends ReactContextBaseJavaModule {
+    public static final String MODULE_NAME = "ScreenBrightness";
+
+    private static final String PERMISSION_EVENT_NAME = "screenBrightnessPermission";
+    private static final int BRIGHTNESS_MAX = 255;
+    private static final int BRIGHTNESS_MIN = 0;
+
+    private final int mWriteSettingsRequestCode;
```

Comment:
What approach do you when you're setting this to a function argument representing the same thing? Are you guys just using `this.x`?

```java
Activity activity;
TheConstructor (Activity activity) {
  this.activity = activity;
}
```